packages/blob/src/put.ts

```diff
@@ -0,0 +1,125 @@
+import type { Readable } from 'node:stream';
+import type { BodyInit } from 'undici';
+import { fetch } from 'undici';
+import type { ClientPutCommandOptions } from './client';
+import {
+  BlobAccessError,
+  BlobUnknownError,
+  getApiUrl,
+  getApiVersionHeader,
+  getTokenFromOptionsOrEnv,
+  BlobError,
+} from './helpers';
+import { type PutCommandOptions } from '.';
+
+const putOptionHeaderMap = {
+  cacheControlMaxAge: 'x-cache-control-max-age',
+  addRandomSuffix: 'x-add-random-suffix',
+  contentType: 'x-content-type',
+};
+
+export interface PutBlobResult {
```

Comment:
just an idea for a separate PR...it would be nice to consume these interfaces from the worker, to ensure client and server uses the same fields

Reply:
Indeed, I have been reluctant to do so because then it ties both services very tightly (for example: you cannot easily add a new property to edge-functions unless you update the npm package).